json-rpc/src/methods.rs

```diff
@@ -243,14 +245,25 @@ async fn get_state_proof(
     StateProofView::try_from((request.ledger_info, proofs.0, proofs.1))
 }
 
+/// Returns the account state to the client, alongside a proof relative to the version and
+/// ledger_version specified by the client. If version or ledger_version are not specified,
+/// the latest known versions will be used.
 async fn get_account_state_with_proof(
     service: JsonRpcService,
     request: JsonRpcRequest,
 ) -> Result<AccountStateWithProofView> {
     let address: String = serde_json::from_value(request.get_param(0))?;
     let account_address = AccountAddress::from_str(&address)?;
-    let version: u64 = serde_json::from_value(request.get_param(1))?;
-    let ledger_version: u64 = serde_json::from_value(request.get_param(2))?;
+    let mut version: u64 = request.version();
+    let mut ledger_version: u64 = request.version();
+
+    // If versions are specified by the request parameters, overwrite the defaults
+    if let Ok(param_version) = serde_json::from_value::<u64>(request.get_param(1)) {
```

Comment:
nit: maybe a bit more idiomatic to use unwrap_or here
`let version = serde_json::from_value::<u64>(request.get_param(1)).unwrap_or(request.version())` ?

Reply:
Ooh, tasty, thanks! Note: I've gone for unwrap_or_else, as request.version() is a function call :) See: https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or